app/search/gleaner.py

```diff
@@ -263,6 +268,33 @@ def text_search(self, **kwargs):
         self.query = GleanerSearch.build_query(user_query, None, page_number)
         return self.execute_query(page_number)
 
+    # A helper method to escape char for different matches
+    def escape_char(self,text):
+        count = text.count('"') #for " " (double quotes)
+        count2 = text.count("'") #for ' ' (for single quotes)
+        
+        #if it has a quote and the count of quotes is odd add an extra quote e.g "Biobasis " Zackenberg" (double quotes)
+        if text.startswith('"') and text.endswith('"') and count%2==1 :
```

Comment:
Also, does it matter whether the text starts and ends with quotes? They can be anywhere, as long as there's an even number of them right? Your tests cases look good, so you could try taking out the checks for starting and ending quotes, and see if you still get the correct results.

Reply:
I just tried removing the starts with and ewnds with and it still works.
I would commit the new changes very soon